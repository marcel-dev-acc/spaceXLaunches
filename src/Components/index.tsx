export { default as Background } from "./Background/Background";
export { default as FilterButton } from "./FilterButton/FilterButton";
export { default as LaunchItem } from './LaunchItem/LaunchItem';
export { default as LoadingModal } from './LoadingModal/LoadingModal';
export { default as Logo } from './Logo/Logo';
export { default as ReloadButton } from './ReloadButton/ReloadButton';
export { default as SortButton } from './SortButton/SortButton';