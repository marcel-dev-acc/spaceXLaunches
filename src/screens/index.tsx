export {default as HomeScreen} from './Home/Home';
export {default as LaunchesScreen} from './Launches/Launches';
