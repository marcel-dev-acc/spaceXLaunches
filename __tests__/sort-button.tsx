import 'react-native';
import React from 'react';
import { SortButton } from '../src/components';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('render sort button', () => {
  const setSortDirectionAsc = (sortDirectionAsc: boolean) => {};
  const sortLaunchesAsc = () => {};
  const sortLaunchesDsc = () => {};
  const tree: any = renderer.create(<SortButton
    sortDirectionAsc={true}
    setSortDirectionAsc={setSortDirectionAsc}
    sortLaunchesAsc={sortLaunchesAsc}
    sortLaunchesDsc={sortLaunchesDsc}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
