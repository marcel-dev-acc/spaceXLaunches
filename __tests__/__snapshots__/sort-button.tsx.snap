// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render sort button 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      false,
      false,
      Object {
        "backgroundColor": "rgba(61, 96, 170, 1)",
        "margin": 5,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingBottom": 0,
        "paddingLeft": 10,
        "paddingTop": 0,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "400",
            "letterSpacing": 0.15,
            "lineHeight": 24,
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "writingDirection": "ltr",
          },
          Object {
            "alignSelf": "center",
            "color": "rgba(255, 255, 255, 1)",
          },
        ]
      }
    >
      Sort Ascending
    </Text>
    <View
      collapsable={false}
      style={
        Object {
          "alignSelf": undefined,
          "bottom": undefined,
          "left": undefined,
          "position": undefined,
          "right": undefined,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "top": undefined,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": undefined,
              "borderColor": "rgba(121, 116, 126, 1)",
              "borderRadius": 18,
              "borderWidth": 0,
              "elevation": 0,
              "height": 36,
              "margin": 0,
              "overflow": "hidden",
              "width": 36,
            }
          }
        >
          <View
            accessibilityComponentType="button"
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessibilityTraits="button"
            accessible={true}
            centered={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                false,
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              accessibilityElementsHidden={true}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                ]
              }
            >
              â–¡
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
